src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/ProjectSystem/RemoteProjectSnapshot.cs

```diff
@@ -156,6 +161,17 @@ private RazorConfiguration CreateRazorConfiguration()
             razorLanguageVersion = RazorLanguageVersion.Latest;
         }
 
-        return new(razorLanguageVersion, configurationName, Extensions: [], UseConsolidatedMvcViews: true);
+        var compilation = await _project.GetCompilationAsync().ConfigureAwait(false);
+
+        var suppressAddComponentParameter = compilation is null
+            ? false
+            : !compilation.HasAddComponentParameter();
```

Comment:
nit: This might read a little better with the boolean logic flipped.

```suggestion
        var suppressAddComponentParameter = compilation is not null
            ? compilation.HasAddComponentParameter()
            : true;
```